core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/functions.scala

```diff
@@ -12,299 +12,329 @@ case class CallFunction(function_name: String, arguments: Seq[Expression]) exten
   override def prettyName: String = function_name.toUpperCase(Locale.getDefault)
 }
 
+/**
+ * Extractor object used to match instances of `CallFunction` in a case insensitive manner. The following pattern `case
+ * CaseInsensitiveCallFunction("FOO", args) => ...` will match `CallFunction("foo", Seq(...))` as well as
+ * `CallFunction("FOO", Seq(...))`, etc.
+ *
+ * The function name specified in the pattern must be all uppercase for this to work.
+ */
+object CaseInsensitiveCallFunction {
+  def unapply(call: CallFunction): Option[(String, Seq[Expression])] = {
+    Some((call.function_name.toUpperCase(), call.arguments))
+  }
+}
+
 class CallMapper {
 
   /** This function is supposed to be overridden by dialects */
   def convert(call: Fn): Fn = call match {
```

Comment:
```suggestion
  def convert(call: Fn): Fn = withNormalizedName(call) match {
```
this diff is too big and unnecessary. just add a private method to fix the name.

Reply:
It's only big because it fixes a big chunk of code that was introduced without tests and bypassing code review.